dynamodb.go

```diff
@@ -107,3 +162,32 @@ func (db dynamoDBSerivce) WaitForReadyTable() error {
 
 	return nil
 }
+
+func (db dynamoDBSerivce) Scan(items chan []DynamoDBItem, totalSegments, segment int) error {
```

Comment:
```suggestion
func (db dynamoDBSerivce) Scan(itemsChan chan<- []DynamoDBItem, totalSegments, segment int) error {
```

(optional) I find it weird to receive a channel here whereas the `BatchWrite` func receives an array of `DynamoDBItem`. Maybe receive a `func(items []DynamoDBItem)` instead would make it more flexible in case this function gets re-used in a context that doesn't use channels.